File: docs/test_button_pot.py
from machine import Pin, ADC
import time

BUTTON_PIN = 2
LED_PIN = 3
POT_PIN = 26   # ADC0 no Pico

button = Pin(BUTTON_PIN, Pin.IN, Pin.PULL_UP)
led = Pin(LED_PIN, Pin.OUT)
pot = ADC(POT_PIN)

blink_mode = False
last_button_state = 1
last_blink_time = time.ticks_ms()
led_output_state = False

print("MicroPython Test Started")

def map_range(value, in_min, in_max, out_min, out_max):
    return int((value - in_min) * (out_max - out_min) / (in_max - in_min) + out_min)

while True:
    # --- 1. Botão com debounce ---
    current_button_state = button.value()

    if current_button_state == 0 and last_button_state == 1:
        time.sleep_ms(30)  # debounce
        if button.value() == 0:
            blink_mode = not blink_mode
            print("Blink Mode Enabled" if blink_mode else "Blink Mode Disabled")

    last_button_state = current_button_state

    # --- 2. LED comportamento ---
    if not blink_mode:
        led.value(1)
        led_output_state = True  # garante estado consistente

    else:
        pot_value = pot.read_u16()  # 0–65535

        # Mapeia para 200–2000 ms
        periodo = map_range(pot_value, 0, 65535, 200, 2000)
        half_period = periodo // 2

        now = time.ticks_ms()

        if time.ticks_diff(now, last_blink_time) >= half_period:
            last_blink_time = now
            led_output_state = not led_output_state
            led.value(led_output_state)

    time.sleep_ms(5)

File: poc/tests/test_button_pot.cpp
// Button + Potentiometer Blink Control (C++)
// Button on D2, LED on D3 (PWM), Potentiometer on A0

const int BUTTON_PIN = 2;
const int LED_PIN = 3;
const int POT_PIN = A0;

bool blinkMode = false; // false = LED aceso, true = pisca
bool lastButtonState = HIGH;

unsigned long lastBlinkTime = 0;
bool ledOutputState = false;

void setup() {
  pinMode(BUTTON_PIN, INPUT_PULLUP);
  pinMode(LED_PIN, OUTPUT);
  Serial.begin(9600);
  Serial.println("Blink + Pot Test Started");
}

void loop() {
  // 1. Button Toggle Logic
  bool currentButtonState = digitalRead(BUTTON_PIN);
  if (currentButtonState == LOW && lastButtonState == HIGH) {
    blinkMode = !blinkMode;
    Serial.println(blinkMode ? "Blink Mode Enabled" : "Blink Mode Disabled");
  }
  lastButtonState = currentButtonState;

  // 2. LED Behavior
  if (!blinkMode) {
    // --- LED ACESO FIXO ---
    digitalWrite(LED_PIN, HIGH);
  } else {
    // --- LED PISCANDO COM VELOCIDADE DO POTENCIÔMETRO ---

    int potValue = analogRead(POT_PIN);

    // Pot 0% → 200 ms (5 Hz)
    // Pot 100% → 2000 ms (0.5 Hz)
    int periodo = map(potValue, 0, 1023, 200, 2000);
    int halfPeriod = periodo / 2;

    unsigned long now = millis();

    if (now - lastBlinkTime >= halfPeriod) {
      lastBlinkTime = now;
      ledOutputState = !ledOutputState;
      digitalWrite(LED_PIN, ledOutputState);
    }

    // Debug
    static int counter = 0;
    if (++counter > 50) {
      Serial.print("Pot: ");
      Serial.print(potValue);
      Serial.print(" | Periodo: ");
      Serial.print(periodo);
      Serial.print(" ms | LED: ");
      Serial.println(ledOutputState ? "ON" : "OFF");
      counter = 0;
    }
  }

  delay(5);
}
